pkg/kp/consulutil/safe.go

```diff
@@ -2,10 +2,71 @@ package consulutil
 
 import (
 	"errors"
+	"fmt"
+	"path/filepath"
+	"runtime"
+
+	"github.com/square/p2/pkg/util"
+	"github.com/square/p2/pkg/util/param"
 
 	"github.com/square/p2/Godeps/_workspace/src/github.com/hashicorp/consul/api"
 )
 
+// Show detailed error messages from Consul (use only when debugging)
+var showConsulErrors = param.Bool("show_consul_errors_unsafe", false)
+
+// KVError encapsulates an error in a Store operation. Errors returned from the
+// Consul API cannot be exposed because they may contain the URL of the request,
+// which includes an ACL token as a query parameter.
+type KVError struct {
+	Op          string
+	Key         string
+	UnsafeError error
+	filename    string
+	function    string
+	lineNumber  int
+}
+
+var _ util.CallsiteError = KVError{}
```

Comment:
Actually, I'm going to get rid of this assertion.  `"util"` isn't otherwise included.